test/signalfd/main.d

```diff
@@ -436,7 +436,19 @@ private class SignalFDTest
         // FLAKY: call to epoll_wait() may fail or return wrong number of events
         test!("!=")(epoll_res, -1); // epoll_wait() error
         test!("==")(epoll_res, 1); // one event fired
-        assert(fired_events[0].data.obj is signal_fd, "unexpected event data");
+
+        // We need to handle if the test is built with EpollFdSanity
+        // as we're using epoll event directly, which may have registered
+        // fd stuffed in into the highest byte.
+        debug (EpollFdSanity)
```

Comment:
Huh, it's annoying to have to bear this in mind for any test which might use epoll directly. The only alternative would be to not run this test at all, in `-debug=EpollFdSanity`, which I don't think is an especially better option. What do you think? (I'm especially considering future tests that may be added, that would also need to bear this in mind.)

Reply:
I'll have to think a bit more about this. Somehow this doesn't taste good - there's asymmetry in the `Epoll`'s API now - registration doesn't match return values. I think what I'll need to do is to not embed registration protection this inside `epoll.ctl`, but to add new method into `Epoll` (maybe `epoll.safeCtl`), which will be used by `EpollSelectDispatcher` (and others, but then they'll know they need to handle the event parsing afterwards.

Reply:
Hmm, other thing to do is to handle this safety check inside `EpollSelectDispatcher` and on the other side `SelectedKeysHandler`, and not touch `Epoll` at all. I think that might be the best approach.

Reply:
Yes, that would be good to try.